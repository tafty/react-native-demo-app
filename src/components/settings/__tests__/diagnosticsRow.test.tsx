import 'react-native'
import React from 'react'
import { render } from 'react-native-testing-library'

import DiagnosticsRow from '../diagnosticsRow'

describe('DiagnosticsRow', () => {
  it('should render the row', () => {
    const title = 'The title'
    const value = 'The value'

    const { getByTestId, toJSON } = render(
      <DiagnosticsRow title={title} value={value} />,
    )

    expect(getByTestId('The title-title').props.children).toBe(title)
    expect(getByTestId('The title-value').props.children).toBe(value)

    expect(toJSON()).toMatchSnapshot()
  })
})
