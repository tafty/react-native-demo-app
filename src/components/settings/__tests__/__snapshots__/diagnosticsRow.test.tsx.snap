// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiagnosticsRow should render the row 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "rgba(246, 246, 246, 1)",
      "borderBottomWidth": 1,
      "flex": 0,
      "flexDirection": "column",
      "justifyContent": "space-between",
      "padding": 8,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 16,
      }
    }
    testID="The title-title"
  >
    The title
  </Text>
  <Text
    style={
      Object {
        "fontSize": 16,
        "fontWeight": "bold",
        "textAlign": "right",
      }
    }
    testID="The title-value"
  >
    The value
  </Text>
</View>
`;
